app/src/main/java/com/babylon/wallet/android/presentation/transfer/TransferViewModel.kt

```diff
@@ -611,11 +663,13 @@ sealed class SpendingAsset {
     abstract val resourceAddress: ResourceAddress
     abstract val resourceAddressOrGlobalId: String
     abstract val isValidForSubmission: Boolean
+    abstract val canDeposit: Boolean
```

Comment:
apart from that some addresses are modelled as Strings, you probably should wrap all `Boolean` as well. As to not accidentally use the `isValidForSubmission` for `exceedingBalance` or `canDeposit` etc, as seen in https://github.com/radixdlt/sargon/pull/199/files#diff-b9958a3c0f355085a78e9f789e925ea5f34011b26522bae686e660553d5e3bd4R37-R39 by @matiasbzurovski

Reply:
This can become a bit cumbersome in the viewmodels, and **probably** it is not necessary most of times. And I explain why:
- This is the state of the viewmodel, it's the final (destination) layer to the UI, we should ensure that the road from the data layer and then through the domain layer is type safe and unwrap the values in the state of the viewmodel
- we can easily test the state by creating previews in composable, and we should do! This way we can eagerly and easily see if a val was accidentally used.

Reply:
but wrap these `val`s in sargon makes totally sense!